Added support of CryptoNight v8 Waltz and CryptoNight v8 ReverseWaltz

EDDragonWolf opened the pull request:
…) and CryptoNight v8 ReverseWaltz (aka cryptonight-rwz or rwz)

mbg033 commented on xmrig/common/xmrig.h:
```diff
@@ -61,20 +61,22 @@ enum AlgoVariant {
 
 
 enum Variant {
-    VARIANT_AUTO = -1, // Autodetect
```
seems like extra spaces added for new lines (between constant name t and '=' sign). Maybe remove to have minimal set of the changes?

EDDragonWolf replied:
Yeah, I thought about it, but I tried to keep alignment since `VARIANT_WALTZ` is longer than other on one symbol. I think, we can go with it, but if maintainers say to remove, I will do it, maybe I'll rename WALTZ to WLZ or WTZ.

MoneroOcean committed:
Restore fixed seed usage

MoneroOcean committed:
Fixed rx seed size

MoneroOcean committed:
Moved initial VM init out of the perf loop

MoneroOcean committed:
Added RX Loki support info

MoneroOcean committed:
Simplified tests

MoneroOcean committed:
Refactored for xmrig v3.0.0

MoneroOcean committed:
Fixed ARM compile

MoneroOcean committed:
Moved to xmrig mem alloc

MoneroOcean committed:
Moved to xmrig mem alloc

MoneroOcean committed:
Added new rx algo

MoneroOcean committed:
Added new rx algo

MoneroOcean committed:
Added new rx algo

MoneroOcean committed:
Fixed error

MoneroOcean committed:
Fixed error

MoneroOcean committed:
Fixed error

MoneroOcean committed:
Fixed error

MoneroOcean committed:
Fixed error

MoneroOcean committed:
Fixed error

MoneroOcean committed:
Fixed error

MoneroOcean committed:
Fixed error

MoneroOcean committed:
Fixed error

MoneroOcean committed:
Fixed error

MoneroOcean committed:
Fixed error

MoneroOcean committed:
Fixed error

MoneroOcean committed:
Fixed error

MoneroOcean committed:
Fixed error

MoneroOcean committed:
Fixed error

MoneroOcean committed:
Fixed error

MoneroOcean committed:
Fixed error

MoneroOcean committed:
Fixed error

MoneroOcean committed:
Fixed error

MoneroOcean committed:
Fixed error

MoneroOcean committed:
Fixed error

MoneroOcean committed:
Fixed error

MoneroOcean committed:
Fixed error

MoneroOcean committed:
Fixed error

MoneroOcean committed:
Fixed error

MoneroOcean committed:
Fixed error

MoneroOcean committed:
Moved version

MoneroOcean committed:
Added new algos

MoneroOcean committed:
Fixed xmrig override

MoneroOcean committed:
Added argon 3dparty

MoneroOcean committed:
Added argon 3dparty

MoneroOcean committed:
Added argon 3dparty

MoneroOcean committed:
Added argon 3dparty

MoneroOcean committed:
Added argon 3dparty

MoneroOcean committed:
Added tests

MoneroOcean committed:
Fixed bug

MoneroOcean committed:
Fixed ARM

MoneroOcean committed:
Optimized Intel/AMD

MoneroOcean committed:
Fixed error

MoneroOcean committed:
Removed useless hash

MoneroOcean committed:
Fixed error with AMD compilation

MoneroOcean committed:
Moved version

MoneroOcean committed:
Restored fix for AMR32

MoneroOcean committed:
Moved version

MoneroOcean committed:
Possible fix for non AES CPU

MoneroOcean committed:
Possible fix for non AES CPU

MoneroOcean committed:
Possible fix for non AES CPU

MoneroOcean committed:
Updated version

MoneroOcean committed:
Added K12 algo support

MoneroOcean committed:
Added K12 tests

MoneroOcean committed:
Added K12 tests

MoneroOcean committed:
Fix K12 C binding

MoneroOcean committed:
Fixed input format

MoneroOcean committed:
Added K12 perf test

MoneroOcean committed:
Moved version

MoneroOcean committed:
Adjusted iteration number

MoneroOcean committed:
Updated from v4.2.0-beta xmrig

MoneroOcean committed:
Fixed rx/0 tests

MoneroOcean committed:
Restored fixes

MoneroOcean committed:
Moved version

MoneroOcean committed:
Changes from xmrig v4.3.0-beta

MoneroOcean committed:
Changes from xmrig v4.3.0-beta

MoneroOcean committed:
Changes from xmrig v4.3.0-beta

MoneroOcean committed:
Changes from xmrig v4.3.0-beta

MoneroOcean committed:
Changes from xmrig v4.3.0-beta

MoneroOcean committed:
Changes from xmrig v4.3.0-beta

MoneroOcean committed:
Changes from xmrig v4.3.0-beta

MoneroOcean committed:
Changes from xmrig v4.3.0-beta

MoneroOcean committed:
Changes from xmrig v4.3.0-beta

MoneroOcean committed:
Changes from xmrig v4.3.0-beta

MoneroOcean committed:
Changes from xmrig v4.3.0-beta

MoneroOcean committed:
Moved version

MoneroOcean committed:
Added RX/ARQ tests

MoneroOcean committed:
Added ArqMa

Snipa22 committed:
Multihashing-Fix: Support older machines

Older OS installs do not correctly bootstrap the stdexcept library.
This breaks the throw std::domain_error("Unknown RandomX algo");
As the std::domain_error is not imported.

MoneroOcean committed:
Merge pull request #38 from Snipa22/patch-1

Multihashing-Fix: Support older machines

Commit:
RandomV

MoneroOcean committed:
Added XMV test

MoneroOcean committed:
Updated utils

MoneroOcean committed:
Updated utils

MoneroOcean committed:
c29s/c29v support

MoneroOcean committed:
c29s/c29v support

MoneroOcean committed:
Fixed package.json

MoneroOcean committed:
Fixed bug

MoneroOcean committed:
Fixed bug

MoneroOcean committed:
Fixed bug

MoneroOcean committed:
Fixed bug

MoneroOcean committed:
Fixed bug

MoneroOcean committed:
Fixed undefined symbol: blake2b

MoneroOcean committed:
Fixed undefined symbol: blake2b

MoneroOcean committed:
Moved version

bobbieltd committed:
Node v12 support

I just use same way from cn-util repo's commit.

kevacoin-project committed:
Added Keva.

MoneroOcean committed:
Dero support

MoneroOcean committed:
Bug fix

MoneroOcean committed:
Bug fix

MoneroOcean committed:
Bug fix

MoneroOcean committed:
Bug fix

MoneroOcean committed:
Bug fix

MoneroOcean committed:
Bug fix

MoneroOcean committed:
Bug fix

MoneroOcean committed:
Bug fix

MoneroOcean committed:
Bug fix

MoneroOcean committed:
Bug fix

MoneroOcean committed:
Bug fix

MoneroOcean committed:
Bug fix

MoneroOcean committed:
Bug fix

MoneroOcean committed:
Bug fix

MoneroOcean committed:
Bug fix

MoneroOcean committed:
Bug fix

MoneroOcean committed:
Fixed AstroBWT

MoneroOcean committed:
Fixed AstroBWT

MoneroOcean committed:
Improved test

MoneroOcean committed:
Fixed AstroBWT

MoneroOcean committed:
Fixed AstroBWT

MoneroOcean committed:
Merge pull request #40 from bobbieltd/patch-2

Node v12 support - Tested on my pool.

MoneroOcean committed:
Merge pull request #44 from kevacoin-project/master

Please add RandomKeva

MoneroOcean committed:
Added rx/keva algo support

MoneroOcean committed:
Fix for v12

MoneroOcean committed:
Fix for v12

MoneroOcean committed:
Fix for v12

MoneroOcean committed:
Update for xmrig v6.2.0-beta

MoneroOcean committed:
Bug fixes

MoneroOcean committed:
Bug fixes

MoneroOcean committed:
Bug fixes

MoneroOcean committed:
Bug fixes

MoneroOcean committed:
Bug fixes

MoneroOcean committed:
Bug fixes

MoneroOcean committed:
Bug fixes

MoneroOcean committed:
Bug fixes

MoneroOcean committed:
Bug fixes

MoneroOcean committed:
Bug fixes

MoneroOcean committed:
Bug fixes

MoneroOcean committed:
Bug fixes

MoneroOcean committed:
Revert "Bug fixes"

This reverts commit 61d8852df776748133f94517a0ea2fc59423b19d.

MoneroOcean committed:
Bug fixes

MoneroOcean committed:
Bug fixes

MoneroOcean committed:
Improved tests

MoneroOcean committed:
Added CCX support

MoneroOcean committed:
Updated version

MoneroOcean committed:
Added CCX support

MoneroOcean committed:
Bug fixes

MoneroOcean committed:
KawPow experimental implementation

MoneroOcean committed:
Bug fixes

MoneroOcean committed:
Bug fixes

MoneroOcean committed:
Bug fixes

MoneroOcean committed:
Bug fixes

MoneroOcean committed:
Bug fixes

MoneroOcean committed:
Bug fixes

MoneroOcean committed:
Bug fixes

MoneroOcean committed:
Bug fixes

MoneroOcean committed:
Bug fixes

MoneroOcean committed:
Bug fixes

MoneroOcean committed:
Fixed test

MoneroOcean committed:
Bug fixes

MoneroOcean committed:
Bug fixes

MoneroOcean committed:
Bug fixes

MoneroOcean committed:
Bug fixes

MoneroOcean committed:
Bug fixes

MoneroOcean committed:
Bug fixes

MoneroOcean committed:
Bug fixes

MoneroOcean committed:
Bug fixes

MoneroOcean committed:
Bug fixes

MoneroOcean committed:
Bug fixes

MoneroOcean committed:
Bug fixes

MoneroOcean committed:
Bug fixes

MoneroOcean committed:
Returned version back

tubedev2000 committed:
c29b support

MoneroOcean committed:
Merge pull request #46 from BitTubeCash/master

c29b support

MoneroOcean committed:
Moved version

MoneroOcean committed:
Added Panther algo support

MoneroOcean committed:
Bug fix

MoneroOcean committed:
Bug fix

MoneroOcean committed:
Bug fix

MoneroOcean committed:
Bug fix

MoneroOcean committed:
Bug fix

MoneroOcean committed:
Bug fix

MoneroOcean committed:
Bug fix

MoneroOcean committed:
Bug fix

ilie1988 committed:
add c29i XTA support

MoneroOcean committed:
Merge pull request #47 from italocoin-project/master

add c29i XTA support

MoneroOcean committed:
Added new c29i algo

MoneroOcean committed:
xmrig v6.4.0-mo2 merge

MoneroOcean committed:
xmrig v6.4.0-mo2 merge

MoneroOcean committed:
xmrig v6.4.0-mo2 merge

MoneroOcean committed:
xmrig v6.4.0-mo2 merge

MoneroOcean committed:
xmrig v6.4.0-mo2 merge

MoneroOcean committed:
xmrig v6.4.0-mo2 merge

MoneroOcean committed:
xmrig v6.4.0-mo2 merge

MoneroOcean committed:
xmrig v6.4.0-mo2 merge

MoneroOcean committed:
xmrig v6.4.0-mo2 merge

MoneroOcean committed:
xmrig v6.4.0-mo2 merge

MoneroOcean committed:
xmrig v6.4.0-mo2 merge

MoneroOcean committed:
xmrig v6.4.0-mo2 merge

MoneroOcean committed:
Updated Ravencoin to use faster verify

MoneroOcean committed:
Updated Ravencoin to use faster verify

MoneroOcean committed:
Updated Ravencoin to use faster verify

MoneroOcean committed:
Updated Ravencoin to use faster verify

MoneroOcean committed:
Updated Ravencoin to use faster verify

MoneroOcean committed:
Updated Ravencoin to use faster verify

MoneroOcean committed:
Updated Ravencoin to use faster verify

MoneroOcean committed:
Updated Ravencoin to use faster verify

MoneroOcean committed:
Updated Ravencoin to use faster verify

MoneroOcean committed:
Updated Ravencoin to use faster verify

MoneroOcean committed:
Fixed kawpow hash verify

MoneroOcean committed:
Fixed kawpow hash verify

MoneroOcean committed:
Fixed kawpow hash verify

MoneroOcean committed:
Fixed kawpow hash verify

MoneroOcean committed:
Ethash support

MoneroOcean committed:
Ethash support

MoneroOcean committed:
Ethash support

MoneroOcean committed:
Ethash support

MoneroOcean committed:
Ethash support

MoneroOcean committed:
Ethash support

MoneroOcean committed:
Ethash support

MoneroOcean committed:
Ethash support

MoneroOcean committed:
Eth light mining support

MoneroOcean committed:
Eth light mining support

MoneroOcean committed:
Eth light mining support

MoneroOcean committed:
Eth light mining support

MoneroOcean committed:
Eth light mining support

MoneroOcean committed:
Eth light mining support

MoneroOcean committed:
Eth light mining support

MoneroOcean committed:
Eth light mining support

MoneroOcean committed:
Eth light mining support

MoneroOcean committed:
Eth light mining support

MoneroOcean committed:
Eth light mining support

MoneroOcean committed:
Eth light mining support

MoneroOcean committed:
Eth light mining support and reversed kawpow result hash

sdbondi committed:
Added mac build support to node-gyp bindings

MoneroOcean committed:
Merge pull request #49 from tari-project/platform-support

Add mac build support to node-gyp bindings

MoneroOcean committed:
Fixed awk call

thundo committed:
Fix Linux AMD New check

Suppress output to avoid
```
make: Entering directory '/myproject/node_modules/cryptonight-hashing/build'
make: Warning: File 'cryptonight-hashing.target.mk' has modification time 0.56 s in the future
  CC(target) Release/obj.target/cryptonight-hashing/xmrig/crypto/cn/asm/cn_main_loop.o
cc: error: vendor_id: No such file or directory
cc: error: :: No such file or directory
cc: error: AuthenticAMD: No such file or directory
cc: error: model: No such file or directory
cc: error: name: No such file or directory
cc: error: AMD: No such file or directory
cc: error: Ryzen: No such file or directory
cc: error: 7: No such file or directory
cc: error: 2700X: No such file or directory
cc: error: Eight-Core: No such file or directory
cc: error: Processor: No such file or directory
cryptonight-hashing.target.mk:251: recipe for target 'Release/obj.target/cryptonight-hashing/xmrig/crypto/cn/asm/cn_main_loop.o' failed
make: *** [Release/obj.target/cryptonight-hashing/xmrig/crypto/cn/asm/cn_main_loop.o] Error 1
make: Leaving directory '/myproject/node_modules/cryptonight-hashing/build'
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:194:23)
gyp ERR! stack     at ChildProcess.emit (events.js:315:20)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:277:12)
gyp ERR! System Linux 5.4.72-microsoft-standard-WSL2
gyp ERR! command "/usr/bin/node" "/usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /myproject/node_modules/cryptonight-hashing
```
during compilation.

thundo committed:
Merge pull request #1 from thundo/fix-linux-amd-check

Fix Linux AMD New check

MoneroOcean committed:
Merge pull request #50 from thundo/master

Fix Linux AMD New check

MoneroOcean committed:
Added autolykos2 support

MoneroOcean committed:
Added autolykos2 support

MoneroOcean committed:
Added autolykos2 support
